File: src/lessons/recursion/hanoi/TricolorHanoi1Entity.scala
package lessons.recursion.hanoi;

import lessons.recursion.hanoi.universe.HanoiEntity;

class TricolorHanoi1Entity extends HanoiEntity {

	override def run() {
    val src = getParamInt(0)
    val mid = getParamInt(1)
    val dst = getParamInt(2)
		move3(getSlotSize(src)/3, src,mid, dst)
	}

	/* BEGIN TEMPLATE */
  def move3(height:Int, src:Int, mid:Int, dst:Int) {
	  /* BEGIN SOLUTION */
    if (height>0) {
//      System.err.println("move3("+height+","+src+","+dst+")");
      move3(height-1, src, dst, mid);
      move(src,dst);
      move(src,dst);
      move(src,dst);
      move3(height-1, mid, src, dst);
    }
		/* END SOLUTION */
	}
	/* END TEMPLATE */
}

File: src/lessons/recursion/hanoi/TricolorHanoi1Entity.py
from RemoteHanoi import *

def run():
    # BEGIN TEMPLATE
    def move3(height, src, mid, dst):
        # Your code here
        # BEGIN SOLUTION
        if (height>0):
            move3(height-1, src, dst, mid)
            move(src,dst)
            move(src,dst)
            move(src,dst)
            move3(height-1, mid, src, dst)
        # END SOLUTION
    # END TEMPLATE
    move3(getSlotSize(getParamInt(0))//3, getParamInt(0),getParamInt(1),getParamInt(2))
